### ISSUE TITLE
Adding validation check to prevent creating two columns with different cases in schema

### ISSUE THREAD

**Issue by fx19880617** [apache/incubator-pinot#6644]
Although pinot columns can be case-sensitive, I think we should still prevent the scenarios like this.
This easily breaks the external case-insensitive integrations like Presto.

E.g. in the same schema, we shouldn't allow creating two columns like: `memberId` and `MemberID`.

**Comment by mcvsubbu** [comment]
I am fine for adding all sorts of validations, but I thought we agreed (perhaps verbally) on doing things in a compatible manner by introducing the concept of a `level`. If the level is set to "strict" then we can do all stuff. 
This change, in particular, is backwards incompatible. It is impossible to check thousands of tables for each condition.
cc: @kishoreg , @icefury71

**Comment by icefury71** [comment]
Yes agreed - this one in particular seems to be backwards incompatible. Currently, we don't provide a mechanism to enforce different levels of validation. We should probably get that done before adding such checks.

**Comment by icefury71** [comment]
@mcvsubbu created #6645 which should be done before addressing this.

**Comment by Vamsi-klu** [comment]
Opened PR to address this: https://github.com/apache/pinot/pull/19089
